torchani/data/new.py

```diff
@@ -0,0 +1,458 @@
+        self.data_coordinates = []
+        self.data_energies = []
+        self.data_self_energies = []
+
+        anidata = anidataloader(file_path)
+        anidata_size = anidata.group_size()
+        enable_pkbar = anidata_size > 5 and PKBAR_INSTALLED
+        if enable_pkbar:
+            pbar = pkbar.Pbar('=> loading h5 dataset into cpu memory, total molecules: {}'.format(anidata_size), anidata_size)
+
+        for i, molecule in enumerate(anidata):
+            num_conformations = len(molecule['coordinates'])
+            # species and coordinates
+            self.data_coordinates += list(molecule['coordinates'].reshape(num_conformations, -1).astype(np.float32))
+            species = np.array([species_dict[x] for x in molecule['species']])
+            self.data_species += list(np.tile(species, (num_conformations, 1)))
+            # energies
+            self.data_energies += list(molecule['energies'].reshape((-1, 1)).astype(np.float32))
+            if subtract_self_energies:
+                self_energies = np.array(sum([self_energies_dict[x] for x in molecule['species']]))
+                self.data_self_energies += list(np.tile(self_energies, (num_conformations, 1)).astype(np.float32))
+            # other properties
+            if enable_pkbar:
+                pbar.update(i)
+
+        if subtract_self_energies:
+            self.data_energies = np.array(self.data_energies) - np.array(self.data_self_energies)
+            import gc
+            del self.data_self_energies
+            del self_energies
+            gc.collect()
+
+        self.batch_size = batch_size
+        self.length = (len(self.data_species) + self.batch_size - 1) // self.batch_size
+        self.device = device
+
+        self.shuffled_index = np.arange(len(self.data_species))
+        np.random.shuffle(self.shuffled_index)
+
+        self.chunk_threshold = chunk_threshold
+        if not self.chunk_threshold:
+            self.chunk_threshold = np.inf
+
+        anidata.cleanup()
+        import gc
+        del num_conformations
+        del species
+        del anidata
+        gc.collect()
+
+    @functools.lru_cache(maxsize=None)
+    def __getitem__(self, index):
+
+        if index >= self.length:
+            raise IndexError()
+
+        batch_indices = slice(index * self.batch_size, (index + 1) * self.batch_size)
+        batch_indices_shuffled = self.shuffled_index[batch_indices]
+
+        batch_species = [self.data_species[i] for i in batch_indices_shuffled]
+        batch_coordinates = [self.data_coordinates[i] for i in batch_indices_shuffled]
+        batch_energies = [self.data_energies[i] for i in batch_indices_shuffled]
+
+        datas = [batch_species, batch_coordinates, batch_energies]
+
+        # get sort index
+        num_atoms_each_mole = [b.shape[0] for b in batch_species]
+        atoms = torch.tensor(num_atoms_each_mole, dtype=torch.int32)
+        sorted_atoms, sorted_atoms_idx = torch.sort(atoms, descending=False)
```

Comment:
`descending` is `False` by default. don't write it again.

Reply:
Sorry, the reader like me don't know the defalut, and I don't want to google to check it.

Reply:
It is a common sense that, in most libraries, if not specified, sorting will be ascending. Now you know it, then change it.